### PR TITLE
bump k8s, remove k8s-e2e

### PR DESCRIPTION
**Pull request by stevekuznetsov**

### CODE REVIEW

**Comment by ncdc** [comment]
@stevekuznetsov need to fix:
```
Error: pkg/sharding/apiserver/delegating_storage.go:123:24: undefined: "k8s.io/kubernetes/pkg/genericcontrolplane".ParseClusterName
Error: pkg/sharding/apiserver/delegating_storage.go:128:14: assignment mismatch: 2 variables but 1 value
```

File: pkg/sharding/apiserver/delegating_storage.go [hunk 1]
```diff
@@ -120,10 +119,7 @@ func (s *delegatingStorage) routedRequest(ctx context.Context, resourceVersion *
 	if err != nil {
 		return nil, nil, err
 	}
-	identifier, _, err := genericcontrolplane.ParseClusterName(clusterName)
-	if err != nil {
-		return nil, nil, err
-	}
+	identifier := "fake"
```
Comment (ncdc) [inline pkg/sharding/apiserver/delegating_storage.go:122]: Is this more of a "FIXME get it compiling for now, come back to it later"?

Comment (stevekuznetsov) [reply pkg/sharding/apiserver/delegating_storage.go:122]: Yeah, I am just turning this off for now, will replace in a follow-up with #353

File: pkg/server/server.go [hunk 2]
```diff
@@ -478,7 +478,7 @@ func (s *Server) Run(ctx context.Context) error {
 		},
 		// user is a virtual cluster that is lazily instantiated
 		"user": {
-			Server:                   server.LoopbackClientConfig.Host + "/clusters/" + genericcontrolplane.SanitizedClusterName(server.ExternalAddress, "user"),
+			Server:                   server.LoopbackClientConfig.Host + "/clusters/" + "user",
```
[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply pkg/server/server.go]: done :)

**Comment by ncdc** [comment]
Linting timed out. Trying again. May need to up timeout beyond 5m if this happens more frequently

**Pull request merged**